core/server/src/main/java/alluxio/master/file/options/CreateDirectoryOptions.java

```diff
@@ -59,8 +59,9 @@ public CreateDirectoryOptions(CreateDirectoryTOptions options) throws IOExceptio
       mGroup = SecurityUtils.getGroupFromThriftClient();
     }
     if (options.isSetMode()) {
-      mDefaultMode = false;
       mMode = new Mode(options.getMode());
+    } else {
+      mMode.applyDirectoryUMask();
```

Comment:
NIT: instead of `applyDirectoryUMask`, how about having `defaultDirectoryMode` and `defaultFileMode`?